solutions/minwoo/baekjoon/16234 인구이동.py

```diff
@@ -0,0 +1,46 @@
+from collections import deque
+n,l,r = map(int,input().split())
+board = [list(map(int,input().split())) for _ in range(n)]
+t = 0
+
+def bfs(y,x): # return: 연합국가갯수, 총인구합, 해당위치좌표들
+    q = deque()
+    visit[y][x] = 1
+    q.append([y,x])
+    pos = [[y,x]]
+    population = board[y][x]
+    region = 1
+    while q :
+        y,x = q.popleft()
+        for dy,dx in ((0,1),(0,-1),(1,0),(-1,0)):
+            ny = y+dy
+            nx = x+dx
+            if 0<=ny<n and 0<=nx<n and not visit[ny][nx] and l<=abs(board[ny][nx] - board[y][x])<=r:
+                visit[ny][nx] = 1
+                q.append([ny,nx])
+                pos.append([ny,nx])
+                population+=board[ny][nx]
+                region+=1
+    return region,population,pos
+
+def move(move_info):
+    for _move_info in move_info :
+        region,population,pos = _move_info
+        carry = population // region # 배치시킬 인원
+        for y,x in pos :
+            board[y][x] = carry
+
+while True :
+    visit = [[0]*n for _ in range(n)]
+    move_info = []
+    for i in range(n):
+        for j in range(n):
+            if not visit[i][j]:
+                region,population,pos = bfs(i,j)
+                if region > 1 :
```

Comment:
```suggestion
                if len(pos) > 1 :
```
코드를 보니 이 부분에 나오는 region을 이 코드로 대체하면 bfs 함수의 리턴으로 region을 받을 게 아니라 carry를 계산해서 받고 region을 리턴값에서 빼도 될 것 같아요. 큰 차이는 없긴 한데 보통 함수의 리턴값이나 전달 인자 수가 적을수록 가독성 측면에서 더 좋다는 이야기를 많이 하거든요 ㅎㅎ 그래봐야 2개 3개 차이니까 큰 차이는 아니긴 합니다. 전체적으로 코드가 보기 좋고 깔끔한 거 같아요 코드 잘 봤습니다!! 👍

Reply:
리뷰 감사합니다.ㅎㅎ 불필요한 리턴값이 있었네요~! 👍